### PR TITLE
Add parameter ReleaseType in RunVersionController.ps1 to support LTS release

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Description
Changes:
- Add parameter ReleaseType for RunVersionController.ps1
- Continue bumping if a version is occupied.
- If bumping version for LTS, 
  - only check same major versions from PSGallery.
  - Change the version of az.accounts in other modules from MinimalVersion to RequiredVersion
- Enable daily-build pipeline for building LTS.
- Several bug fixes  

Test:
- For Az 12 stage, bumping version for STS and LTS should be same, verified by
  - STS build: https://dev.azure.com/azclitools/internal/_build/results?buildId=181621&view=results
  - LTS build: https://dev.azure.com/azclitools/internal/_build/results?buildId=181622&view=results
- Assume bumping version from Az 11, bumping version for LTS should in patch positions (except Az.Accounts), verified by 
  - https://dev.azure.com/azclitools/internal/_build/results?buildId=181585&view=results
  - The reason why Az.Accounts is wrongly identifying dynamic parameters as a minor change: "EnableTestCoverage", "TestCoverageLocation", "TargetName" 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: tools/Tools.Common/Loaders/MetadataLoader.cs [hunk 1]
```diff
@@ -30,17 +30,21 @@ namespace Tools.Common.Loaders
 {
     public class MetadataLoader
     {
+        private static string _rootPath = Path.GetFullPath(Path.Combine(Assembly.GetExecutingAssembly().Location, "..", "..", ".."));
```
Comment (dolauli) [inline tools/Tools.Common/Loaders/MetadataLoader.cs:33]: What is the difference between Assembly.GetExecutingAssembly().Location and AppDomain.CurrentDomain.BaseDirectory?

File: tools/Tools.Common/Loaders/MetadataLoader.cs [hunk 2]
```diff
@@ -30,17 +30,21 @@ namespace Tools.Common.Loaders
...
+
         public static ModuleMetadata GetModuleMetadata(string moduleName)
         {
-            string rootPath = Path.GetFullPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..", ".."));
-            string modulePsd1Path = Directory.GetFiles(Path.Combine(rootPath, "artifacts"), $"{moduleName}.psd1", SearchOption.AllDirectories)[0];
+            // bez: notice that this search way always find artifacts/Debug/{moduleName}/psd1 first, which may cause some issues
```
Comment (dolauli) [inline tools/Tools.Common/Loaders/MetadataLoader.cs:37]: Shall we need to add these two lines comments in our docs to maker scrum master to be aware of that?

File: tools/Tools.Common/Models/CmdletMetadata.cs [hunk 3]
```diff
@@ -354,12 +354,17 @@ public override bool Equals(Object obj)
                 var otherParameterSet = other.ParameterSets.Find(p => string.Equals(p.Name, thisParameterSet.Name, StringComparison.OrdinalIgnoreCase));
                 if (otherParameterSet == null)
                 {
+                    // Console.WriteLine($"Parameter set {thisParameterSet.Name} in cmdlet {this.Name} is not found in new module.");
```
Comment (dolauli) [inline tools/Tools.Common/Models/CmdletMetadata.cs:357]: Please remove those comment-out code.

File: tools/Tools.Common/Models/ParameterSetMetadata.cs [hunk 4]
```diff
@@ -58,13 +62,24 @@ public override bool Equals(Object obj)
...
                 {
+                    // Console.WriteLine($"Parameter {thisParameter.ParameterMetadata.Name} in parameter set {this.Name} is not found in new module.");
                     return false;
                 }
-
                 paramsSetEqual &= thisParameter.Equals(otherParameter);
             }
 
-            paramsSetEqual &= this.Parameters.Count == other.Parameters.Count;
+            var curParameters = this.Parameters.Where(p => !CommonInfo.ExcludedParameters.Contains(p.ParameterMetadata.Name)).ToList();
```
Comment (dolauli) [inline tools/Tools.Common/Models/ParameterSetMetadata.cs:71]: Is line#71~line#82 necessary?

**Review by dolauli — COMMENTED** [review]
Overall, LGTM

File: tools/Tools.Common/Loaders/MetadataLoader.cs (hunk 1 above)
Comment (BethanyZhou) [reply tools/Tools.Common/Loaders/MetadataLoader.cs:33]: Assembly.GetExecutingAssembly().Location is always azure-powershell\artifacts\VersionController\VersionController.Netcore.dll. AppDomain.CurrentDomain.BaseDirectory is the workspace of powershell.

File: tools/Tools.Common/Loaders/MetadataLoader.cs (hunk 2 above)
Comment (BethanyZhou) [reply tools/Tools.Common/Loaders/MetadataLoader.cs:37]:
```
Build code
Clean the project.

<# Avoid issue of filename too long #>
git config --global core.longpaths true

<# Wipe any changes from a previous branch #>
git clean -xdf
```
Good suggestion. In our docs, we will clean all files under artifacts folder before build code. do you need me to add reasons here?

File: tools/Tools.Common/Models/CmdletMetadata.cs (hunk 3 above)
Comment (BethanyZhou) [reply tools/Tools.Common/Models/CmdletMetadata.cs:357]: I don't want to remove them as it's useful if we need to debug version bumper in ADO pipeline.

File: tools/Tools.Common/Models/CmdletMetadata.cs [hunk 5]
```diff
@@ -354,12 +354,17 @@ public override bool Equals(Object obj)
...
                     return false;
                 }
 
                 cmdletsEqual &= thisParameterSet.Equals(otherParameterSet);
             }
 
+            if (this.ParameterSets.Count != other.ParameterSets.Count)
```
Comment (Nickcandy) [inline tools/Tools.Common/Models/CmdletMetadata.cs:364]: should we remove these code ?

File: tools/RunVersionController.ps1 [hunk 6]
```diff
@@ -194,15 +198,20 @@ function Bump-AzVersion
 {
     Write-Host "Getting local Az information..." -ForegroundColor Yellow
     $localAz = Import-PowerShellDataFile -Path "$PSScriptRoot\Az\Az.psd1"
-
-    Write-Host "Getting gallery Az information..." -ForegroundColor Yellow
-    $galleryAz = Find-Module -Name Az -Repository $GalleryName
+    Write-Host $localAz.ModuleVersion
```
Comment (Nickcandy) [inline tools/RunVersionController.ps1:201]:
```suggestion

```

File: tools/VersionController/Program.cs [hunk 7]
```diff
@@ -47,45 +49,73 @@ public class Program
...
+                }
+            }           
+        }
+
+        private static void GenerateSyntaxChangelog(string projectDirectories)
+        {
+            try
             {
-                _moduleNameFilter = args[1] + Psd1NameExtension;
+                _syntaxChangelogGenerator.Analyze(projectDirectories);
```
Comment (Nickcandy) [inline tools/VersionController/Program.cs:112]: remove syntax changelog process in LTS

File: tools/RunVersionController.ps1 [hunk 8]
```diff
@@ -313,7 +320,7 @@ function Update-AzPreview
         $Psd1Object = Import-PowerShellDataFile $Psd1FilePath
         $moduleName = [System.IO.Path]::GetFileName($Psd1FilePath) -replace ".psd1"
         $moduleVersion = $Psd1Object.ModuleVersion.ToString()
-        if('Az.Accounts' -eq $moduleName)
+        if('Az.Accounts' -eq $moduleName -and "STS" -eq $ReleaseType)
```
Comment (Nickcandy) [inline tools/RunVersionController.ps1:323]: Should skip bump up az-preview in LTS

**Review by Nickcandy — APPROVED** [review]

**Pull request merged**